src/lib/model_selection/_split.ts

```diff
@@ -208,3 +210,188 @@ export function train_test_split(
     yTrain: clean(yTrain),
   };
 }
+
+const rangeValidationError = (type, size, n_samples) => `${type}=${size} should be either 
+positive and smaller than number of samples ${n_samples} or a float in (0, 1) range`;
+
+const testRangeValidationError = (test_size, n_samples) => rangeValidationError('test_size', test_size, n_samples);
+
+const trainRangeValidationError = (test_size, n_samples) => rangeValidationError('test_size', test_size, n_samples);
+
+function uniq(array: Type1DMatrix<any> = null): Type2DMatrix<number> {
```

Comment:
I think the name `uniq` will easily be confused with lodash `uniq`, so I'd rename this to something else. Could you please think if you can refactor this function to use tfjs APIs instead of raw JS array APIs?

Reply:
If you think this function might be used in other places, move to utils/tensors =)